src/pydvl/valuation/methods/msr_banzhaf.py

```diff
@@ -0,0 +1,201 @@
+"""
+This module implements the MSR-Banzhaf valuation method, as described in
+(Wang et. al.)<sup><a href="wang_data_2023">3</a></sup>.
+
+## References
+
+[^1]: <a name="wang_data_2023"></a>Wang, J.T. and Jia, R., 2023.
+    [Data Banzhaf: A Robust Data Valuation Framework for Machine Learning](https://proceedings.mlr.press/v206/wang23e.html).
+    In: Proceedings of The 26th International Conference on Artificial Intelligence and Statistics, pp. 6388-6421.
+
+"""
+from __future__ import annotations
+
+import warnings
+
+import numpy as np
+from joblib import Parallel, delayed
+from typing_extensions import Self
+
+from pydvl.utils.progress import Progress
+from pydvl.valuation.dataset import Dataset
+from pydvl.valuation.methods.semivalue import SemivalueValuation
+from pydvl.valuation.result import ValuationResult
+from pydvl.valuation.samplers import MSRSampler
+from pydvl.valuation.stopping import StoppingCriterion
+from pydvl.valuation.types import ValueUpdateKind
+from pydvl.valuation.utility.base import UtilityBase
+from pydvl.valuation.utils import (
+    ensure_backend_has_generator_return,
+    make_parallel_flag,
+)
+
+__all__ = ["MSRBanzhafValuation"]
+
+
+class MSRBanzhafValuation(SemivalueValuation):
+    """Class to compute Maximum Sample Re-use (MSR) Banzhaf values.
+
+    See [Data Valuation][data-valuation] for an overview.
+
+    The MSR Banzhaf valuation approximates the Banzhaf valuation and is much more
+    efficient than traditional Montecarlo approaches.
+
+    Args:
+        utility: Utility object with model, data and scoring function.
+        sampler: Sampling scheme to use. Currently, only one MSRSampler is implemented.
+            In the future, weighted MSRSamplers will be supported.
+        is_done: Stopping criterion to use.
+        progress: Whether to show a progress bar.
+
+    """
+
+    algorithm_name = "MSR-Banzhaf"
+
+    def __init__(
+        self,
+        utility: UtilityBase,
+        sampler: MSRSampler,
+        is_done: StoppingCriterion,
+        progress: bool = True,
+    ):
+        sampler = MSRSampler()
```

Comment:
That is quite unexpected, because the user has to provide a sampler, but that instance is never used. As no message is logged, this could be highly confusing. What is the reason to instantiate a `MSRSampler` here?

Reply:
Very good catch. That was a leftover from when I thought there would only be one MSR sampler. But in the future we might add weighted ones. Removed it.